system/library/razorpay-lib/createwebhook.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+require_once __DIR__.'/../razorpay-sdk/Razorpay.php';
+require_once __DIR__.'/../log.php';
+use Razorpay\Api\Api;
+use Razorpay\Api\Errors;
+
+class CreateWebhook
+{
+    protected $keyId = null;
+    protected $keySecret = null;
+
+    protected $webhookId = null;
+    protected $webhookUrl = null;
+    protected $webhookEnable = '1';
+    protected $webhookSecret = null;
+
+    protected $log = null;
+
+    protected $webhookSupportedEvents = [
+        'payment.authorized',
+        'payment.failed',
+        'order.paid',
+        'subscription.paused',
+        'subscription.resumed',
+        'subscription.cancelled',
+        'subscription.charged'
+    ];
+
+    protected $webhookEvents = [
+        'payment.authorized' => true,
+        'payment.failed'     => true,
+        'order.paid'         => true,
+    ];
+
+    function __construct($keyId, $keySecret, $webhookSecret, $webhookUrl)
+    {
+        $this->keyId = $keyId;
+        $this->keySecret = $keySecret;
+        $this->webhookUrl = $webhookUrl;
+
+        if(empty($webhookSecret) === true)
+        {
+            $this->webhookSecret = $this->createWebhookSecret();
+        }
+        else
+        {
+            $this->webhookSecret = $webhookSecret;
+        }
+
+        $this->log = new Log('error.log');
+    }
+
+    public function autoCreateWebhook()
+    {
+        $api = $this->getApiIntance();
+
+        $domain = parse_url($this->webhookUrl, PHP_URL_HOST);
+        $domain_ip = gethostbyname($domain);
```

Comment:
@abdulwahidsharief , please use camelcase.

Reply:
@ramth05 changed to $domainIp